### ISSUE TITLE
Missing PySoundFile dependency: "save() got an unexpected keyword argument 'encoding'"

### ISSUE THREAD

**Issue by isaacbrodsky** [facebookresearch/demucs#358]
## 🐛 Bug Report

The PySoundFile dependency is not installed when the latest demucs is installed via pip. This is the same issue as #302.

## To Reproduce

```
pip install demucs
demucs -d cpu some_wave_file.wav
```

## Expected behavior

The application completes successfully

## Actual Behavior

The following error:
```
100%|████████████████████████████████████████████████████████████████████████| 297.0/297.0 [02:50<00:00,  1.74seconds/s]
Traceback (most recent call last):
  File "C:\...\demucsvenv\Scripts\demucs-script.py", line 11, in <module>
    load_entry_point('demucs==3.0.4', 'console_scripts', 'demucs')()
  File "c:\...\demucsvenv\lib\site-packages\demucs\separate.py", line 159, in main
    save_audio(source, stem, **kwargs)
  File "c:\...\demucsvenv\lib\site-packages\demucs\audio.py", line 254, in save_audio
    encoding=encoding, bits_per_sample=bits_per_sample)
TypeError: save() got an unexpected keyword argument 'encoding'
```

Running `pip install PySoundFile`, which installed version `0.9.0.post1`, and then rerunning the command worked.

## Your Environment

- Python and PyTorch version: Python 3.7.8, pip 20.1.1 (from Visual Studio). PyTorch 1.12.1
- Operating system and version (desktop or mobile): Windows 10
- Hardware (gpu or cpu, amount of RAM etc.): CPU

**Comment by CarlGao4** [comment]
This is a problem quite difficult to solve. 
First of all, `pySoundFile` is deprecated and you should use `soundfile` instead. 
After that, the `TypeError` is new in pyTorch 1.12 because you will get an warning saying that no audio backend is available if you downgrade pyTorch to 1.11
The problem is that pyTorch use `soundfile` to read and write on Windows, but `sox_io` is used on macOS and Linux, requiring users to install sox manually with their system package manager. So adding soundfile to requirements has no good for macOS and Linux users. 
Maybe this can be added to instructions?

**Comment by isaacbrodsky** [comment]
Thanks for pointing that out. I did see a "no audio backend" warning when PySoundFile was not installed.

It sounds like perhaps dependencies (https://peps.python.org/pep-0508/) could be added that are OS specific, like `soundfile; platform_system=="Windows"`, `sox_io; platform_system!="Windows"`?

**Comment by CarlGao4** [comment]
It seems that it should be: 
```
soundfile>=0.10.3;sys_platform=="win32"
sox_io;sys_platform!="win32"
```

Let me do more research

**Comment by mianowill** [comment]
Can confirm `python3 -m pip install soundfile` fixes this with Python 3.10 on Windows 10
